sklearn/feature_selection/_sequential.py

```diff
@@ -145,8 +152,13 @@ def fit(self, X, y):
                      "number of features, or a float in (0, 1] "
                      "representing a percentage of features to "
                      f"select. Got {self.n_features_to_select}")
+
         if self.n_features_to_select is None:
-            self.n_features_to_select_ = n_features // 2
+            # for backwards compability
+            self.n_features_to_select = 'auto'
```

Comment:
This should be 'warn'

And then if n_features_to_select is 'warn', we should add a warning like

>  Leaving n_features_to_select to None is deprecated in 1.0 and will become 'auto' in 1.2.
    To keep the same behaviour as with `None` (i.e. select half of the features) and avoid this warning, you should manually set
    `n_features_to_select='auto'` and set `tol=None`.

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20145/commits/9b11fbbcfb1b443063a25940f4f8d3f5a4c6951e